webapp/src/test/java/rocks/metaldetector/web/controller/rest/UserAccountRestControllerIntegrationTest.java

```diff
@@ -0,0 +1,177 @@
+import rocks.metaldetector.testutil.DtoFactory.UpdateUserRequestFactory;
+import rocks.metaldetector.web.transformer.UserDtoTransformer;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static rocks.metaldetector.support.Endpoints.Rest.USERS;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class UserAccountRestControllerIntegrationTest extends BaseSpringBootTest {
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @MockBean
+  @SuppressWarnings("unused")
+  private UserService userService;
+
+  @MockBean
+  @SuppressWarnings("unused")
+  private UserDtoTransformer userDtoTransformer;
+
+  @Nested
+  class AdministratorRoleTest {
+
+    @Test
+    @DisplayName("Administrator is allowed to GET on endpoint " + USERS + "' (all users)")
+    @WithMockUser(roles = {"ADMINISTRATOR"})
+    void administrator_is_allowed_to_fetch_all_users() throws Exception {
+      mockMvc.perform(get(USERS))
+          .andExpect(status().isOk());
+    }
+
+    @Test
+    @DisplayName("Administrator is allowed to GET on endpoint " + USERS + "' (certain user)")
+    @WithMockUser(roles = {"ADMINISTRATOR"})
+    void administrator_is_allowed_to_fetch_certain_user() throws Exception {
+      mockMvc.perform(get(USERS + "/1"))
+          .andExpect(status().isOk());
+    }
+
+    @Test
+    @DisplayName("Administrator is allowed to POST on endpoint " + USERS + "'")
+    @WithMockUser(roles = {"ADMINISTRATOR"})
+    void administrator_is_allowed_to_create_another_administrator() throws Exception {
+      mockMvc.perform(post(USERS)
+              .with(csrf())
+              .content(objectMapper.writeValueAsString(RegisterUserRequestFactory.createDefault()))
+              .contentType(APPLICATION_JSON)
+          )
+          .andExpect(status().isCreated());
+    }
+
+    @Test
+    @DisplayName("Administrator is allowed to PUT on endpoint " + USERS + "'")
+    @WithMockUser(roles = {"ADMINISTRATOR"})
+    void administrator_is_allowed_to_update_certain_user() throws Exception {
+      mockMvc.perform(put(USERS)
+              .with(csrf())
+              .content(objectMapper.writeValueAsString(UpdateUserRequestFactory.createDefault()))
+              .contentType(APPLICATION_JSON)
+          )
+          .andExpect(status().isOk());
+    }
+  }
+
+  @Nested
+  class UserRoleTest {
+
+    @Test
+    @DisplayName("User is allowed to GET on endpoint " + USERS + "' (all users)")
```

Comment:
```suggestion
    @DisplayName("User is not allowed to GET on endpoint " + USERS + "' (all users)")
```

See also other tests in this class for role user.